### PR TITLE
Update combiner-explainer.md 

### PR DESCRIPTION
**Pull request by sreecha**
The combiner document has references to `workqueue` which is an outdated concept that is replaced by 'executors'. The PR fixes the documentation

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* One or more commits are not associated with a GitHub user. The email used in the Git commit must match a verified email address in a GitHub user profile. For more information, please see https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

_Regards,
CLA GitHub bot_

**Comment by sreecha** [comment]
I forgot..Where do I sign a Contributor License agreement?

**Comment by sreecha** [comment]
I ammended my last commit by updating the author info. The CLA check passes now.

**Comment by nicolasnoble** [comment]
@vjpai: ping

File: doc/core/combiner-explainer.md [hunk 1]
```diff
@@ -114,40 +114,30 @@ if we know that it is valid to do so:
...
 }
 ```
 
-`offload` is more than `break`; it does `break` but also causes some
-other thread that is currently waiting on a poll to break out of its
-poll. This is done by setting up a per-polling-island work-queue
-(distributor) wakeup FD. The work-queue is the converse of the combiner; it
-tries to spray events onto as many threads as possible to get as much concurrency as possible.
+`queue_offload` detaches the combiner from the current thread's executor and
+ offloads to the work to `executor` which is an internal pool of threads.
```
Comment (vjpai) [inline doc/core/combiner-explainer.md]: offloads which work?

Comment (vjpai) [reply doc/core/combiner-explainer.md]: Also, should probably be "offloads the work to " rather than "offloads _to_ the work to"

Comment (sreecha) [reply doc/core/combiner-explainer.md]: Should I phrase it as _"offloads the remaining work to the `executor`"_  ? or is the term "work" ambiguous?

If so, I can rewrite it the following way:

_"Offloads the remaining list of closures (that were scheduled on this combiner) to the `executor`"_

**Comment by vjpai** [comment]
Hmm, not sure this text is still relevant now that the execution model keeps changing. @yashykt : is this still relevant?

**Comment by yashykt** [comment]
Yep, the internals of the combiner haven't changed since then.

**Pull request closed without merging**